### DISCUSSION TITLE
create a clickable container

### DISCUSSION THREAD

**Discussion by oswaldB** [Budibase/budibase#2256] [Ideas]
Hello,

It could be very convenient and very useful to have the capacity to create a clickable container :)

Is it something hard for you to design?

Have a nice day

**Comment by joebudi** [comment]
What happens when you click the container?

**Reply by oswaldB** [reply]
It could have the same options has the button (navigate, execute workflow, and so on)

**Reply by joebudi** [reply]
What's your use case? Adding a simple navigation URL to the container is different from adding multiple actions.

**Reply by oswaldB** [reply]
This way can make an image clickable or a full element.

**Reply by oswaldB** [reply]
I have 2 use cases:
1- I have a picture and when we click on it, it redirects to somewhere else.
2 - I have a full custom card and when I click on it I would like it to send an email.

I am sure many other one use cases can be solved by a container button

**Reply by joebudi** [reply]
Thank you! The first use case seems like a good scenario. The second one, to me, seems a little dangerous - users tend to click screens when reading etc. This might result in a lot of emails getting sent 🙂

**Reply by oswaldB** [reply]
No pb. I control the spaming from the backend :)

**Reply by oswaldB** [reply]
but it can update a row as well

**Reply by joebudi** [reply]
Interesting. And why don't you just use a button?

**Reply by oswaldB** [reply]
I can.

It is mostly an idea that allows us more customization with your tool.

**Reply by joebudi** [reply]
Awesome. Sorry for probing. I wanted to explore the reasoning behind the request. Thanks!

**Reply by oswaldB** [reply]
no worry ;)

**Comment by oswaldB** [comment]
Hello,

here is a new use case for the clickable container : 
![image](https://user-images.githubusercontent.com/3023526/135437050-0a3e0c84-463a-4816-b824-209a61ffe574.png)
I need this to be a button.

**Comment by gvlx** [comment]
Suggestion: add an action setting on a container.
If any action is set then _the container's contents become unreachable_, that is, selecting becomes impossible.

This setting _must not be used if any of the contents are input fields_ (none of the values can be changed).

**Reply by gvlx** [reply]
This should be the default on a `Card`element.

**Comment by samg666** [comment]
I am also looking for clickable container, here are my two use cases.

1. I use repeater on all my project as the table component currently not that flexible, and I would like to have repeater row click action to trigger something like app state update, then trigger my tab-like container data refresh. Currently I have put a button on the repeater row, it work just fine, but it will be much better if it can be trigger on anywhere on the repeater row. A clickable container should help.

<img width="1910" alt="iShot2021-11-26 21 04 46" src="https://user-images.githubusercontent.com/29649651/143585665-1fcd67f8-73d7-416b-935d-c619a62b52b0.png">

2. I have many custom card builded by container, I want my user click the card then trigger app state update, the refresh the repeater. The current solution is put a Button and Icon on the custom card, either click the button or icon will trigger the same action. A clickable container will be better on this use case.
<img width="1621" alt="iShot2021-11-26 21 06 23" src="https://user-images.githubusercontent.com/29649651/143586218-91accb32-65be-4d6b-a898-8f5724d23fe7.png">

**Comment by Zbatux** [comment]
Hello,

I am a fresh user and i wanted to know if anyone did find any way to make images/containers clickable ?

**Comment by dannyshin76** [comment]
Actually we can achieve this, by adding a transparent button with an absolute position that covers its parent container. If you give it a good opacity, you can even make the container look hoverable (which is actually not).

**Reply by melohagan** [reply]
Nice, you're on fire today! I also have a PR out to add the clickable container functionality as well.

**Comment by aderici** [comment]
For anyone else interested, enter this in the CSS portion of the "button" that you will place within your container : 
```position: absolute;
top: 0;
left: 0;
right: 0;
width:100%;
height:100%;
color: rgba(0, 0, 0, 0);
opacity: 0.2;
```

And then you need to add this to the CSS portion of the container itself : 

` position: relative;`

**Comment by gazflo** [comment]
The help file says this is possible, but I can't see were to add the on-click action.

https://docs.budibase.com/docs/container#on-click

**Reply by gazflo** [reply]
Just did a page refresh and now see the option. Strange.
